src/lualib/Error.ts

```diff
@@ -0,0 +1,82 @@
+type TSTLCapturedErrorStack = Array<{
+    namewhat: string;
+    name?: string;
+    source: string;
+    short_src: string;
+    currentline: number;
+}>;
+
+function __TS__GetErrorStack(constructor: Function): TSTLCapturedErrorStack {
+    const functionFrames = [];
+    let level = 1;
+    while (true) {
```

Comment:
This looks dodgy to me, maybe we should set a maximum level limit for this while loop to prevent infinite loops?

Reply:
The loop is already limited by the maximum Lua stack size as `debug.getinfo` [will eventually](https://www.lua.org/pil/23.1.html) return `nil`. We could set an additional limit on the returned stack size, but I'd prefer keeping the full stack.